src/parser/shared/normalizers/PrePullCooldowns.js

```diff
@@ -140,8 +140,14 @@ class PrePullCooldowns extends EventsNormalizer {
          * information more accurately.
          */
         if (precastClassResources === null && event.classResources) {
-          debug && console.debug('Setting prepull class resources to:', event.classResources);
+          debug && console.debug('Setting prepull class resources (but stripping costs) to:', event.classResources);
           precastClassResources = event.classResources;
+          // make a new class resources with costs stripped - have to deep copy to avoid messing with the original cast event
+          precastClassResources = precastClassResources.map(classResource => {
+            const noCostClassResource = Object.assign({}, classResource);
```

Comment:
In the modern JS version we use, you can write this like so:

```
const noCostClassResource ={
  ...classResource,
  cost: undefined,
};
```

With that this entire section could become:

```
precastClassResources = precastClassResources.map((classResource) =>({
  ...classResource,
  cost: undefined,
}))
```

Reply:
Right I forgot you could do that, will change it